iati/core/rulesets.py

```diff
@@ -428,35 +569,44 @@ def is_valid_for(self, dataset):
 
         Raises:
             AttributeError: When an argument is given that does not have the required attributes.
-            XPathEvalError(lxml.etree.XPathEvalError): When no valid XPath is available.
 
         """
-        pattern = re.compile(self.case['regex'])
+        context_elements = self._find_context_elements(dataset)
+        pattern = re.compile(self.regex)
 
-        for path in self.paths:
-            results = dataset.xml_tree.xpath(path)
-            for result in results:
-                return bool(pattern.match(result.text))
+        for context_element in context_elements:
+            if self._evaluate_condition(context_element):
+                return None
+            for path in self.paths:
+                results = context_element.xpath(path)
+                strings_to_check = self._extract_text_from_element_or_attribute(results)
+                for string_to_check in strings_to_check:
+                    if not pattern.match(string_to_check):
```

Comment:
> Python offers two different primitive operations based on regular expressions: re.match() checks for a match only at the beginning of the string, while re.search() checks for a match anywhere in the string (this is what Perl does by default).

https://docs.python.org/2/library/re.html#search-vs-match

Based on this, `search` should probably be used rather than `match` because Rulesets contain `perl style` regexes.

Reply:
The same goes for regexnomatches

Reply:
done